adapters/repos/db/lsmkv/segment_collection_strategy.go

```diff
@@ -26,7 +26,7 @@ func (s *segment) getCollection(key []byte) ([]value, error) {
 			StrategySetCollection, StrategyMapCollection)
 	}
 
-	if !s.bloomFilter.Test(key) {
+	if s.bloomFilter != nil && !s.bloomFilter.Test(key) {
```

Comment:
We should check the settings instead of the nil value.  This has been the source of pervious errors in weaviate, because a mistake initialising the segment means that weaviate will silently continue with the wrong behaviour.  Checking useBloomFilter against the nil value catches errors instead of propogating them.

Reply:
True, that was my initial implementation, but removed it to avoid extending `segment` with `useBloomFilter` field.
Will put field back to the `segment` and replace nil check.